src/python_testing/TC_OpstateCommon.py

```diff
@@ -1024,63 +1043,77 @@ async def TEST_TC_OPSTATE_BASE_2_5(self, endpoint=1):
         # STEP 1: Commission DUT to TH (can be skipped if done in a preceding test)
         self.step(1)
 
-        # STEP 2: Set up a subscription to the OperationCompletion event
-        self.step(2)
-        if self.pics_guard(self.check_pics(f"{self.test_info.pics_code}.S.E01")):
+        # Store PICS value for the Op Complete Event
+        opcomplete_pics = self.pics_guard(self.check_pics(f"{self.test_info.pics_code}.S.E01"))
+
+        # Get Device Types
+        device_type_list = await self.read_single_attribute_check_success(cluster=Clusters.Descriptor,
+                                                                          attribute=Clusters.Descriptor.Attributes.DeviceTypeList,
+                                                                          endpoint=endpoint)
+
+        # Check to see if this test is mandated for the device type in question, that is, we expect support for the OpComplete Event
+        istestmandated = self.opcomplete_test_mandated(device_type_list)
+
+        if istestmandated and not opcomplete_pics:
+            # Device type requires the event, PICS is missing the event, fail
+            asserts.fail("OperationComplete Event support mandated for device type, but not indicated in the PICS")
+
+        # STEP 2: Verify the PICS is set, if not, skip the entire TC. If yes, set up a subscription to the OperationCompletion event
+        if opcomplete_pics:
```

Comment:
silly q, but why not reverse this so you don't have to indent the whole test?
```
if not opcomplete_pics:
  self.skip_all_remaining_steps(2)
  return
```

I don't care that much, tbh, so feel free to close this if you don't fancy another trip through the CI.

Reply:
Reversed the logic